downloader/src/main/kotlin/VersionControlSystem.kt

```diff
@@ -205,16 +205,16 @@ abstract class VersionControlSystem {
         // commit, but the repository still has a tag for the package version (pointing to a different commit). In order
         // to allow to fall back to the guessed revision based on the version in such cases, use a prioritized list of
         // revision candidates instead of a single revision.
-        val revisionCandidates = mutableListOf<String>()
+        val revisionCandidates = mutableSetOf<String>()
 
         try {
             pkg.vcsProcessed.revision.also {
                 if (it.isNotBlank() && (isFixedRevision(workingTree, it) || allowMovingRevisions)) {
-                    log.info {
-                        "Adding $type revision '$it' (taken from package meta-data) as a candidate."
+                    if (revisionCandidates.add(it)) {
```

Comment:
consider eliminating this `if` by adding `&& revisionCandidates.add(it)` to above if condition.

Reply:
I have a slight preference for keeping the nested `if` to highlight this as the code snippet that needs to be used when adding new candidates, like we also do a few lines below.

Reply:
sure.